falcon/response.py

```diff
@@ -384,6 +384,20 @@ def set_header(self, name, value):
         # NOTE(kgriffs): normalize name by lowercasing it
         self._headers[name.lower()] = value
 
+    def delete_header(self, name):
+        """Delete a header for this response.
+
+        If the header was not previously set, do nothing.
+
+        Args:
+            name (str): Header name (case-insensitive).  Must be of type
+                ``str`` or ``StringType`` and contain only US-ASCII characters.
+                Under Python 2.x, the ``unicode`` type is also accepted,
+                although such strings are also limited to US-ASCII.
+        """
+        # NOTE(kgriffs): normalize name by lowercasing it
+        self._headers.pop(name.lower(), None)
```

Comment:
Was curious so I tried benchmarking it myself. If my benchmarks are correct, then my results confirm what you found, @kgriffs. Though it's noteworthy that `try...except` performs much slower in the case when the key in question is not present.

Results:
```
pop_present: 0.9311439991
pop_not_present: 0.939202070236
try_present: 0.687428951263
try_not_present: 4.01770091057
```

My benchmarking code:
```
from timeit import timeit

def pop_present():
    d = {'hello': 'world'}
    d.pop('hello', None)


def pop_not_present():
    d = {'hello': 'world'}
    d.pop('not_present', None)


def try_present():
    d = {'hello': 'world'}
    try:
        del d['hello']
    except KeyError:
        pass


def try_not_present():
    d = {'hello': 'world'}
    try:
        del d['not_present']
    except KeyError:
        pass

N = 5000000
print 'pop_present:', timeit('pop_present()', 'from __main__ import pop_present', number=N)
print 'pop_not_present:', timeit('pop_not_present()', 'from __main__ import pop_not_present', number=N)
print 'try_present:', timeit('try_present()', 'from __main__ import try_present', number=N)
print 'try_not_present:', timeit('try_not_present()', 'from __main__ import try_not_present', number=N)
```